libcxx/include/typeinfo

```diff
@@ -275,7 +275,7 @@ struct __type_info_implementations {
           __impl;
 };
 
-#    if defined(__arm64__) && __has_cpp_attribute(clang::ptrauth_vtable_pointer)
+#    if __has_cpp_attribute(clang::ptrauth_vtable_pointer)
```

Comment:
This should be _Uglified. Also, why is this not in `__config`?

Reply:
Since this is only used for `std::type_info`, I feel it makes more sense to keep this macro localized to `<typeinfo>`. If we used it elsewhere then it would make more sense to lift it to `__config`. WDYT?

Regarding uglifying, the right spelling would be `__has_cpp_attribute(__clang__::__ptrauth_vtable_pointer__)` and we can also change e.g. `__has_feature(__ptrauth_type_info_discriminated_vtable_pointer__)`, right?